python/plugins/processing/algs/qgis/postgis_utils.py

```diff
@@ -153,7 +153,7 @@ def __init__(self, host=None, port=None, dbname=None, user=None,
         try:
             self.con = psycopg2.connect(self.con_info())
         except psycopg2.OperationalError as e:
-            raise DbError(unicode(e))
+            raise DbError(str(e))
```

Comment:
Agree with you. This mask the original error to the user/developper. This is the case in many place in processing plugin, take a look here :

https://github.com/qgis/QGIS/blob/56783c915ed129c2d8dc9132c2bba6b6911a6f46/python/plugins/processing/tools/vector.py#L575:L587

My objective here was to fix SQL error message display, not to refactor processing error handling.